<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from ilhuicamina.xacro              | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="rover">
  <!-- Definimos las dimensiones del lidar-->
  <link name="lidar_link">
    <inertial>
      <mass value="0.0"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
    <collision>
      <geometry>
        <cylinder length="0.04" radius="0.025"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.02"/>
    </collision>
    <visual>
      <geometry>
        <cylinder length="0.04" radius="0.025"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.02"/>
    </visual>
  </link>
  <!-- Definimos el joint entre el lidar y la base -->
  <joint name="base_to_lidar" type="fixed">
    <parent link="base_link"/>
    <child link="lidar_link"/>
    <origin rpy="0 0 0" xyz="0.3 0 0.47"/>
  </joint>
  <!-- Dummy link to act as the root link -->
  <link name="dummy_link">
    <!-- No visual, collision, or inertial elements needed -->
  </link>
  <link name="base_link">
    <pose>0 0 0 0 0 0</pose>
    <inertial>
      <mass value="35.689"/>
      <origin rpy="0 0 0" xyz="0 0 0.085"/>
      <inertia ixx="1.5921292357633332" ixy="0" ixz="0" iyy="0.3900320247741667" iyz="0" izz="0.3900320247741667"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://bring_up/meshes/Chasis.STL" scale="0.001 0.001 .001"/>
      </geometry>
      <origin rpy="1.570796327 0 1.570796327 " xyz="-.365 -.175 0.262"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://bring_up/meshes/Chasis.STL" scale="0.001 0.001 .001"/>
      </geometry>
      <origin rpy="1.570796327 0 1.570796327 " xyz="-.365 -.175 0.262"/>
    </collision>
  </link>
  <!-- Fixed joint connecting dummy_link to base_link -->
  <joint name="fixed_dummy_base" type="fixed">
    <parent link="dummy_link"/>
    <child link="base_link"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
  </joint>
  <!-- Definimos las dimensiones de la camara-->
  <link name="ZED_link">
    <inertial>
      <mass value="0.0"/>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
    <collision>
      <geometry>
        <box size="0.0431 0.17525 0.0325"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
    </collision>
    <visual>
      <geometry>
        <box size="0.0431 0.17525 0.0325"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
    </visual>
  </link>
  <joint name="base_to_ZED" type="fixed">
    <parent link="base_link"/>
    <child link="ZED_link"/>
    <origin rpy="0 0 0" xyz="0.4 0.0 0.38"/>
  </joint>
  <joint name="ZED_depth_joint" type="fixed">
    <origin rpy="-1.570796326795 0 -1.570796326795 " xyz="0 0 0"/>
    <parent link="ZED_link"/>
    <child link="ZED_depth_link"/>
  </joint>
  <link name="ZED_depth_link"/>
  <!--LLANTA IZQUIERDA FRONT-->
  <link name="flwheel_link">
    <inertial>
      <mass value="2.2"/>
      <origin rpy="0 1.570796327 1.570796327" xyz="0 0 0"/>
      <inertia ixx="0.02091375" ixy="0" ixz="0" iyy="0.02091375" iyz="0" izz="0.029947500000000005"/>
    </inertial>
    <visual name="link_flwheel_visual">
      <geometry>
        <mesh filename="package://bring_up/meshes/wheel_left_f.STL"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.034 -0.013 0"/>
    </visual>
    <collision name="link_flwheel_collision">
      <geometry>
        <mesh filename="package://bring_up/meshes/wheel_left_f.STL"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.034 -0.013 0"/>
    </collision>
  </link>
  <joint name="base_to_flwheel" type="continuous">
    <parent link="base_link"/>
    <child link="flwheel_link"/>
    <origin rpy="1.570796327 0 0" xyz="0.386 0.375 0"/>
    <axis rpy="0 0 0" xyz="0 0 -1"/>
    <limit effort="10000" velocity="1000"/>
    <joint_properties damping="1.0" friction="100.0"/>
  </joint>
  <!--LLANTA IZQUIERDA BACK-->
  <link name="blwheel_link">
    <inertial>
      <mass value="2.2"/>
      <origin rpy="0  1.570796327 1.570796327" xyz="0 0 0"/>
      <inertia ixx="0.02091375" ixy="0" ixz="0" iyy="0.02091375" iyz="0" izz="0.029947500000000005"/>
    </inertial>
    <visual name="link_blwheel_visual">
      <geometry>
        <mesh filename="package://bring_up/meshes/wheel_left_f.STL"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.034 -0.013 0"/>
    </visual>
    <collision name="link_blwheel_collision">
      <geometry>
        <mesh filename="package://bring_up/meshes/wheel_left_f.STL"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.034 -0.013 0"/>
    </collision>
  </link>
  <joint name="base_to_alwheel" type="continuous">
    <parent link="base_link"/>
    <child link="blwheel_link"/>
    <origin rpy="1.570796327 0 0" xyz="-0.45 0.375 0"/>
    <axis rpy="0 0 0" xyz="0 0 -1"/>
    <limit effort="10000" velocity="1000"/>
    <joint_properties damping="1.0" friction="100.0"/>
  </joint>
  <!--LLANTA IZQUIERDA ENMEDIO-->
  <link name="middle_wheel">
    <inertial>
      <mass value="2.2"/>
      <origin rpy="0 1.570796327 1.570796327" xyz="0 0 0"/>
      <inertia ixx="0.02091375" ixy="0" ixz="0" iyy="0.02091375" iyz="0" izz="0.029947500000000005"/>
    </inertial>
    <visual name="link_wheel">
      <geometry>
        <mesh filename="package://bring_up/meshes/wheel_left_f.STL"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.034 -0.013 0"/>
    </visual>
    <collision name="link_wheel_collision">
      <geometry>
        <mesh filename="package://bring_up/meshes/wheel_left_f.STL"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.034 -0.013 0"/>
    </collision>
  </link>
  <joint name="base_to_middle_wheel" type="continuous">
    <parent link="base_link"/>
    <child link="middle_wheel"/>
    <origin rpy="1.570796327 0 0" xyz="-0.037 0.375 0"/>
    <axis rpy="0 0 0" xyz="0 0 -1"/>
    <limit effort="10000" velocity="1000"/>
    <joint_properties damping="1.0" friction="100.0"/>
  </joint>
  <!-- LLANTA DERECHA FRONT-->
  <link name="frwheel_link">
    <inertial>
      <mass value="2.2"/>
      <origin rpy="0 1.570796327 1.570796327" xyz="0 0 0"/>
      <inertia ixx="0.02091375" ixy="0" ixz="0" iyy="0.02091375" iyz="0" izz="0.029947500000000005"/>
    </inertial>
    <visual name="link_frwheel_visual">
      <geometry>
        <mesh filename="package://bring_up/meshes/wheel_left_f.STL"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.034 -0.013 0"/>
    </visual>
    <collision name="link_frwheel_collision">
      <geometry>
        <mesh filename="package://bring_up/meshes/wheel_left_f.STL"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.034 -0.013 0"/>
    </collision>
  </link>
  <joint name="base_to_frwheel" type="continuous">
    <parent link="base_link"/>
    <child link="frwheel_link"/>
    <origin rpy="1.570796327 0 0" xyz="0.40 -0.352 0"/>
    <axis rpy="0 0 0" xyz="0 0 -1"/>
    <limit effort="10000" velocity="1000"/>
    <joint_properties damping="1.0" friction="100.0"/>
  </joint>
  <!--LLANTA DERECHA ATRAS-->
  <link name="brwheel_link">
    <inertial>
      <mass value="2.2"/>
      <origin rpy="0 1.570796327 1.570796327" xyz="0 0 0"/>
      <inertia ixx="0.02091375" ixy="0" ixz="0" iyy="0.02091375" iyz="0" izz="0.029947500000000005"/>
    </inertial>
    <visual name="link_brwheel_visual">
      <geometry>
        <mesh filename="package://bring_up/meshes/wheel_left_f.STL"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.034 -0.013 0"/>
    </visual>
    <collision name="link_brwheel_collision">
      <geometry>
        <mesh filename="package://bring_up/meshes/wheel_left_f.STL"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.034 -0.013 0"/>
    </collision>
  </link>
  <joint name="base_to_arwheel" type="continuous">
    <parent link="base_link"/>
    <child link="brwheel_link"/>
    <origin rpy="1.570796327 0 0" xyz="-0.45 -0.352 0"/>
    <axis rpy="0 0 0" xyz="0 0 -1"/>
    <limit effort="10000" velocity="1000"/>
    <joint_properties damping="1.0" friction="100.0"/>
  </joint>
  <!--LLANTA DERECHA ENMEDIO-->
  <link name="middle_wheel_2">
    <inertial>
      <mass value="2.2"/>
      <origin rpy="0 1.570796327 1.570796327" xyz="0 0 0"/>
      <inertia ixx="0.02091375" ixy="0" ixz="0" iyy="0.02091375" iyz="0" izz="0.029947500000000005"/>
    </inertial>
    <visual name="link_wheel_2">
      <geometry>
        <mesh filename="package://bring_up/meshes/wheel_left_f.STL"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.034 -0.013 0"/>
    </visual>
    <collision name="link_wheel_collision_2">
      <geometry>
        <mesh filename="package://bring_up/meshes/wheel_left_f.STL"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.034 -0.013 0"/>
    </collision>
  </link>
  <joint name="base_to_middle_wheel_2" type="continuous">
    <parent link="base_link"/>
    <child link="middle_wheel_2"/>
    <origin rpy="1.570796327 0 0" xyz="-0.037 -0.352 0"/>
    <axis rpy="0 0 0" xyz="0 0 -1"/>
    <limit effort="10000" velocity="1000"/>
    <joint_properties damping="1.0" friction="100.0"/>
  </joint>
</robot>
